src/renderer/contexts/ExecutionContext.tsx

```diff
@@ -328,8 +328,21 @@ export const ExecutionProvider = memo(({ children }: React.PropsWithChildren<{}>
 
     const kill = useCallback(async () => {
         try {
-            const response = await backend.kill();
-            if (response.type === 'error') {
+            // Try to kill the current executor
+            // If it doesn't respond within 1 second, force restart it
+            const backendKillPromise = backend.kill();
+            const timeoutPromise = new Promise<string>((resolve) => {
+                setTimeout(() => {
+                    resolve('timeout');
+                }, 1000);
+            });
```

Comment:
Please use the `delay` function and promise chaining. This not only makes the code simpler, it also results in better types. The `typeof response !== 'string'` below should be unnecessary now.

```suggestion
            const timeoutPromise = delay(1000).then(() => 'timeout');
```

Reply:
Maybe make it adjustable in settings? 5sec feels too long to me, since the point is to stop an upscale instead of waiting for it to complete.

Reply:
Honestly, even just the 1 second feels a bit too long. When i was playing around with it yesterday, since the backend takes time to restart it really feels like nothing happens for quite a while. Maybe we need a modal with a spinner to pop up while it's stopping or something, then we can increase it to like 2.5s